File: bench/spellcheck/spellcheck.lua
-- $Id: spellcheck.lua,v 1.2 2004-05-25 02:26:33 bfulgham Exp $
-- http://www.bagley.org/~doug/shootout/
-- based on code from Roberto Ierusalimschy

assert(io.open("Usr.Dict.Words"))
local dict = {}
for line in io.lines("Usr.Dict.Words") do
  dict[line] = true
end

for word in io.lines() do
  if not dict[word] then print(word) end
end

File: bench/spellcheck/spellcheck.php
#!/usr/local/bin/php -f
<?php
/*
 $Id: spellcheck.php,v 1.1 2004-05-29 00:00:44 bfulgham Exp $
 http://www.bagley.org/~doug/shootout/
*/
$dict = array();
$fd = fopen("Usr.Dict.Words", "r");
while (!feof ($fd)) { $dict[chop(fgets($fd, 1024))] = 1; }
fclose($fd);

$fd = fopen("php://stdin", "r");
while (!feof ($fd)) {
    $word = chop(fgets($fd, 1024));
    if (! $dict[$word]) {
	print "$word\n";
    }
}
fclose($fd);
?>
